src/pomelo_plug.c

```diff
@@ -15,11 +15,144 @@
+    i2 = ((i) - 3)  & mask; \
+    i3 = ((i) - 7)  & mask; \
+    i4 = ((i) - 13) & mask; \
+    S[i0] = ADD256(S[i0], XOR256(ADD256(XOR256(S[i1], S[i2]), S[i3]), S[i4]));      \
+    S[i0] = ROTL256_64(S[i0]);  \
+    S[i0] = ROTL256(S[i0],17); \
+}
+
+// Function G update the state using function F together with Key-INdependent random memory accesses
+#define G(i,random_number)  {                                                       \
+    index_global = (random_number >> 16) & mask;                                    \
+    for (j = 0; j < 32; j++)                                                        \
+    {                                                                               \
+        F(i+j);                                                                     \
+        index_global   = (index_global + 1) & mask;                                 \
+        index_local    = (i + j - 0x1000 + (random_number & 0x1fff)) & mask;        \
+        S[i0]          = ADD256(S[i0],SHIFTL256(S[index_local],1));                 \
+        S[index_local] = ADD256(S[index_local],  SHIFTL256(S[i0],2));               \
+        S[i0]          = ADD256(S[i0],SHIFTL256(S[index_global],1));                \
+        S[index_global]= ADD256(S[index_global], SHIFTL256(S[i0],3));               \
+        random_number += (random_number << 2);                                      \
+        random_number  = (random_number << 19) ^ (random_number >> 45)  ^ 3141592653589793238ULL;   \
+    }                                                                               \
+}
+
+// Function H update the state using function F together with Key-dependent random memory accesses
+#define H(i, random_number)  {                                                      \
+    index_global = (random_number >> 16) & mask;                                    \
+    for (j = 0; j < 32; j++)                                                        \
+    {                                                                               \
+        F(i+j);                                                                     \
+        index_global   = (index_global + 1) & mask;                                 \
+        index_local    = (i + j - 0x1000 + (random_number & 0x1fff)) & mask;        \
+        S[i0]          = ADD256(S[i0],SHIFTL256(S[index_local],1));                 \
+        S[index_local] = ADD256(S[index_local],  SHIFTL256(S[i0],2));               \
+        S[i0]          = ADD256(S[i0],SHIFTL256(S[index_global],1));                \
+        S[index_global]= ADD256(S[index_global], SHIFTL256(S[i0],3));               \
+        random_number  = ((unsigned long long*)S)[(i3 << 2)];                       \
+    }                                                                               \
+}
+
+int PHS_pomelo(void *out, size_t outlen, const void *in, size_t inlen, const void *salt, size_t saltlen, unsigned int t_cost, unsigned int m_cost);
```

Comment:
Unneeded line. I wonder where it came from.

Reply:
It is part of the code from the upstream tarball. I will remove it.